cpp/daal/src/services/service_defines.h

```diff
@@ -71,7 +71,7 @@ DAAL_EXPORT bool daal_check_is_intel_cpu();
     #define PRAGMA_ICC_TO_STR(ARGS)
     #define PRAGMA_ICC_OMP(ARGS)
     #define PRAGMA_ICC_NO16(ARGS)
-    #define DAAL_TYPENAME
+    #define DAAL_TYPENAME typename
```

Comment:
what does this do?

Reply:
Add typename instead DAAL_TYPENAME for some classes, it's necessary for icx and doesnt affect icc